deploy/nginx.conf

```diff
@@ -0,0 +1,152 @@
+# [2]: https://github.com/cloudfoundry/staticfile-buildpack/blob/05867b82df7e5f6f1155ce193a424a79693967bb/src/staticfile/finalize/data.go
+
+worker_processes 1;
+daemon off;
+
+error_log stderr;
+events { worker_connections 1024; }
+
+pid /tmp/nginx.pid;
+
+http {
+  charset utf-8;
+
+  # Configure logging
+
+  log_format cloudfoundry 'NginxLog "$request" $status $body_bytes_sent';
+  access_log /dev/stdout cloudfoundry;
+
+  # MIME types and default MIME types
+
+  default_type application/octet-stream;
+  include nginx/mime.types;
+
+  # Enabling sendfile eliminates copying file data into buffer and sends it
+  # directly
+
+  sendfile on;
+
+  # Enable gzip for text-based resources
+
+  gzip on;
+  gzip_disable "msie6";
+  gzip_comp_level 6;
+  gzip_min_length 1100;
+  gzip_buffers 16 8k;
+  gzip_proxied any;
+  gunzip on;
+  gzip_static always;
+  gzip_types text/plain text/css text/js text/xml text/javascript application/javascript application/x-javascript application/json application/xml application/xml+rss;
+  gzip_vary on;
+
+  # Yeah, like, networking stuff ¯\_(ツ)_/¯
+
+  tcp_nopush on;
+  keepalive_timeout 30;
+
+  # Disable absolute redirects
+  #
+  # The index directive (1) used in the nginx config within the buildpack means
+  # that we send a 301 redirect when visiting a directory without a trailing
+  # slash (e.g. /design-system will redirect to /design-system/).
+  # 
+  # However by default nginx will use absolute URLs – including the hostname –
+  # when redirecting, which means that users going to
+  # design-system.service.gov.uk/components will end up being redirected to our
+  # 'origin domain' (govuk-design-system-production.cloudapps.digital/components/)
+  # which is less than ideal.
+  #
+  # Disabling absolute redirects (2) means that nginx will instead 301 to the
+  # relative url /design-system, which means users won't be redirected away from
+  # the service domain and everything is A-OK ✌️
+  #
+  # [1]: http://nginx.org/en/docs/http/ngx_http_index_module.html#index
+  # [2]: http://nginx.org/en/docs/http/ngx_http_core_module.html#absolute_redirect
+
+  absolute_redirect off;
+
+  # Ensure that redirects don't include the port which nginx is using inside the
+  # CloudFoundry container
+
+  port_in_redirect off;
+
+  # Send different expires headers depending on the content-type of the resource
+  # we're returning. It should be safe to cache CSS, JavaScript and fonts
+  # 'indefinitely' as they should include fingerprints in their filenames.
+  #
+  # Because images and videos are not fingerprinted we only cache them for 30
+  # minutes.
+  # 
+  # The epoch parameter corresponds to the absolute time
+  # 'Thu, 01 Jan 1970 00:00:01 GMT'.
+
+  map $sent_http_content_type $expires {
+    default                         off;
+    text/html                       epoch;
+    text/css                        max;
```

Comment:
I suppose new assets will have a different name but setting max expires makes me somewhat uneasy.

Reply:
What do you suggest?